src/predefined/continuous_famous_systems.jl

```diff
@@ -1405,3 +1498,297 @@ end
+    Analysis, synthesis, labyrinth chaos.
+    Int. J. Bifurc. Chaos.
+"""
+function thomas(u0=[2.199442,2.3634225,3.220197]; a = 1.85, b = 10)
+    return CDS(loop_thomas, u0, [a,b])
+end
+
+thomas_labyrinth(u0=[-4.9599545,1.0302034,-4.6883893]; a=0.5, b=10) = CDS(loop_thomas, u0, [a,b])
+
+function loop_thomas(u, p, t)
+    @inbounds begin
+        a,b = p
+        x,y,z = u
+        du1 = -a * x + b * sin(y)
+        du2 = -a * y + b * sin(z)
+        du3 = -a * z + b * sin(x)
+    end
+    return SVector{3}(du1, du2, du3)
+end
+
+"""
+```julia
+swinging_atwood(u0=[0.113296,1.5707963267948966,0.10992,0.17747]; m1=1.0, m2=4.5)
+```
+```math
+\\begin{aligned}
+\\dot{r} &= \\frac{p_r}{M+m}\\\\
+\\dot{p}_r &= -Mg + mg\\cos(\\theta)\\\\
+\\dot{\\theta} &= \\frac{p_{\\theta}}{mr^2}\\\\
+\\dot{p}_{\\theta} &= -mgr\\sin(\\theta)
+\\end{aligned}
+```
+A mechanical system consisting of two swinging weights connected by ropes and pulleys.
+This is only chaotic when m2 is sufficiently larger than m1, and there are nonzero initial
+momenta [^Tufillaro1984].
+
+[^Tufillaro1984]: 
+    Tufillaro, Nicholas B.; Abbott, Tyler A.; Griffiths, David J. (1984).
+    Swinging Atwood's Machine. American Journal of Physics. 52 (10): 895--903.
+"""
+function swinging_atwood(u0=[0.113296,1.5707963267948966,0.10992,0.17747]; m1=1.0, m2=4.5)
+    return CDS(loop_swingingatwood, u0, [m1, m2])
+end
+
+function loop_swingingatwood(u, p, t)
+    @inbounds begin
+        m1, m2 = p
+        r, th, pr, pth = u
+        g = 9.82
+        du1 = pr/(m1 + m2)
+        du2 = pth/(m1*r^2)
+        du3 = pth^2/(m1*r^3) - m2*g + m1*g*cos(th)
+        du4 = -m1*g*r*sin(th)
+    end
+    return SVector{4}(du1, du2, du3, du4)
+end
+
+"""
+```julia
+guckenheimer_holmes(u0=[-0.55582369,0.05181624,0.37766104]; 
+    a = 0.4,
+    b = 20.25,
+    c = 3,
+    d = 1.6,
+    e = 1.7,
+    f = 0.44)
+```
+```math
+\\begin{aligned}
+\\dot{x} &= ax - by + czx + dz(x^2 + y^2)\\\\
+\\dot{y} &= ay + bx + czy\\\\
+\\dot{z} &= e - z^2 - f(x^2 + y^2) - az^3
+\\end{aligned}
+```
+A nonlinear oscillator [^GuckenheimerHolmes1983].
+
+[^GuckenheimerHolmes1983]:
+    Guckenheimer, John, and Philip Holmes (1983).
+    Nonlinear oscillations, dynamical systems, and bifurcations of vector fields.
+    Vol. 42. Springer Science & Business Media.
+"""
+function guckenheimer_holmes(u0=[-0.55582369,0.05181624,0.37766104]; 
+    a = 0.4,
+    b = 20.25,
+    c = 3,
+    d = 1.6,
+    e = 1.7,
+    f = 0.44)
+    return CDS(loop_guckenheimerholmes, u0, [a,b,c,d,e,f])
+end
+
+function loop_guckenheimerholmes(u, p, t)
+    @inbounds begin
+        a,b,c,d,e,f = p
+        x,y,z = u
+        du1 = a*x - b*y + c*z*x + d*z*(x^2 + y^2)
+        du2 = a*y + b*x + c*z*y
+        du3 = e - z^2 - f*(x^2 + y^2) - a*z^3
+    end
+    return SVector{3}(du1, du2, du3)
+end
+
+"""
+```julia
+halvorsen(u0=[-8.6807408,-2.4741399,0.070775762]; a = 1.4, b = 4.0)
```

Comment:
Where does `halvorsen` come from?

Reply:
chapter 3.6.1 of Sprott's Elegant Chaos: Algebraically Simple Chaotic Flows

> A simple example of such a circulant system was suggested by Arne Dehli Halvorsen (unpublished)